amd_openvx/openvx/api/vx_api.cpp

```diff
@@ -410,12 +410,11 @@ VX_API_ENTRY vx_status VX_API_CALL vxSetContextAttribute(vx_context context, vx_
 #elif ENABLE_HIP
         case VX_CONTEXT_ATTRIBUTE_AMD_HIP_DEVICE:
             if (size == sizeof(hipDevice_t)) {
-                if (context->hip_device < 0 && agoGpuHipCreateContext(context, context->hip_device) != VX_SUCCESS) {
-                    status = VX_FAILURE;
+                if (*(int *)ptr > 0 && agoGpuHipCreateContext(context, *(int *)ptr) == VX_SUCCESS) {
```

Comment:
please use local variable device_id = *(int *)ptr and use that within the if statement